opencti-platform/opencti-front/src/private/components/data/connectors/ConnectorWorks.tsx

```diff
@@ -0,0 +1,376 @@
+import TaskStatus from '../../../../components/TaskStatus';
+import { useFormatter } from '../../../../components/i18n';
+import { FIVE_SECONDS } from '../../../../utils/Time';
+import { commitMutation, MESSAGING$ } from '../../../../relay/environment';
+import Transition from '../../../../components/Transition';
+
+const interval$ = interval(FIVE_SECONDS);
+
+const useStyles = makeStyles(() => ({
+  paper: {
+    height: '100%',
+    minHeight: '100%',
+    margin: '10px 0 0 0',
+    padding: '15px',
+    borderRadius: 6,
+    position: 'relative',
+  },
+  number: {
+    fontWeight: 600,
+    fontSize: 18,
+  },
+  progress: {
+    borderRadius: 5,
+    height: 10,
+  },
+}));
+
+export const connectorWorksWorkDeletionMutation = graphql`
+  mutation ConnectorWorksWorkDeletionMutation($id: ID!) {
+    workEdit(id: $id) {
+      delete
+    }
+  }
+`;
+
+type WorkMessages = NonNullable<NonNullable<NonNullable<ConnectorWorks_data$data['works']>['edges']>[0]>['node']['errors'];
+
+interface Options {
+  count: number,
+  filters: {
+    mode: string,
+    filters: [
+      {
+        key: string
+        values: string[]
+        operator: string
+        mode: string
+      },
+    ],
+    filterGroups: [],
+  },
+}
+
+interface ConnectorWorksComponentProps {
+  data: ConnectorWorks_data$data
+  options: Options[]
+  relay: RelayRefetchProp
+}
+
+const ConnectorWorksComponent: FunctionComponent<ConnectorWorksComponentProps> = ({ data, options, relay }) => {
+  const works = data.works?.edges ?? [];
+  const { t_i18n, nsdt } = useFormatter();
+  const classes = useStyles();
+  const [displayErrors, setDisplayErrors] = useState<boolean>(false);
+  const [errors, setErrors] = useState<WorkMessages>([]);
+
+  const handleOpenErrors = (errorsList: WorkMessages) => {
+    if (!errorsList) return;
+    setDisplayErrors(true);
+    setErrors(errorsList);
+  };
+
+  const handleCloseErrors = () => {
+    setDisplayErrors(false);
+    setErrors([]);
+  };
+
+  // eslint-disable-next-line class-methods-use-this
+  const handleDeleteWork = (workId: string) => {
+    commitMutation({
+      mutation: connectorWorksWorkDeletionMutation,
+      variables: {
+        id: workId,
+      },
+      updater: undefined,
+      optimisticUpdater: undefined,
+      optimisticResponse: undefined,
+      onCompleted: () => {
+        MESSAGING$.notifySuccess('The work has been deleted');
+      },
+      onError: undefined,
+      setSubmitting: undefined,
+    });
+  };
+
+  useEffect(() => {
+    const subscription = interval$.subscribe(() => {
+      relay.refetch(options);
+    });
+    return () => subscription.unsubscribe();
+  }, []);
+
+  return (
+    <div>
+      {works.length === 0 && (
+        <Paper
+          classes={{ root: classes.paper }}
+          variant="outlined"
+          style={{ marginBottom: 20 }}
+        >
+          <div
+            style={{
```

Comment:
It's better to use a `<table>` instead of a `<div>` with a `display: 'table'` 

what do you think ?

Reply:
I'd prefer to simply remove the table, because it's only used here to display the centred `no work` text, and replace it with a Typography component